src/custom/state/price/updater.ts

```diff
@@ -158,15 +158,27 @@ export default function FeesUpdater(): null {
       const refetchPrice = !unsupportedToken && priceIsOld(quoteInfo)
 
       if (unsupportedNeedsCheck || refetchAll || refetchPrice) {
-        setLoadingQuote(true)
+        let quoteData: Pick<QuoteInformationObject, 'chainId' | 'sellToken'> | undefined = undefined
+        // did the trade parameters change? major update required
+        if (quoteInfo && !quoteUsingSameParameters(quoteParams, quoteInfo)) {
+          quoteData = { chainId, sellToken }
+        }
+
+        // here we can set the loading quote state differently
+        // if params changed we signal a major update has happened by passing quoteData
+        // which in state will wipe the price
+        setLoadingQuote({
+          loading: true,
+          quoteData
+        })
 
         refetchQuote({
           quoteParams,
           fetchFee: refetchAll,
           previousFee: quoteInfo?.fee
         })
           .catch(error => console.error('Error re-fetching the quote', error))
-          .finally(() => setLoadingQuote(false))
+          .finally(() => setLoadingQuote({ loading: false }))
```

Comment:
Maybe the `loading: false` is something that should happen automatically in the `updateQuote`?

If you want to also stop the spinner when there's an error, you can let `updateQuote` to receive an error message or make an action for that case. 

This way, if we want, we can even add a small visual indication that last price refresh had an issue...

Reply:
imo feels weird to have loading quote action in 2 different places, but i can expirement

Reply:
handled it in #738 here: https://github.com/gnosis/cowswap/pull/738/files#diff-4a4b1c4981ae5b895e591fc3e25a08661a529b17683e24d7b095e187ea4b3809R167